### ISSUE TITLE
Restrict bundled footer/header patterns to template/template parts

### ISSUE THREAD

**Issue by dsas** [WordPress/gutenberg#45858]
## What problem does this address?

The pattern inserter has several categories which each need expanding to explore. I think the list of categories could be slimmed down. I'm not certain that some of these categories make sense when creating a post and possibly page. Header and Footer are perhaps the obvious candidates, they feel like theme-level patterns not post-level patterns.

![image](https://user-images.githubusercontent.com/93301/202482512-4abcdd90-1d73-4859-93f9-7c4c36444d04.png)

## What is your proposed solution?

In https://github.com/WordPress/gutenberg/issues/41398 a `postTypes` API was added so that patterns could be restricted to certain post types. Gutenberg bundles [a whole bunch of header/footer patterns](https://github.com/WordPress/gutenberg/tree/eb3faf6ffac764ef9b7293750c45fa94d1d6c348/lib/compat/wordpress-6.2/block-patterns) that do not make use of the `postTypes` API. 

We could update all of the bundled header/footer pattern registration calls to include  `postTypes => array( 'wp_template', 'wp_template_part' );`. This would mean those header/footers (and hence the category) would only be available when editing templates and template parts and would not be available when editing pages and posts.

**Comment by richtabor** [comment]
I agree with posts 100%, but for pages - I've seen plenty of use cases where publishers are leveraging the "Blank" template, and creating whole landing page designs (header + content + footer) that are abstracted from the rest of the site.

**Comment by richtabor** [comment]
Perhaps Headers/Footers should be allowed on page post types only?

**Comment by dsas** [comment]
Seems reasonable @richtabor, pages are definitely not as clear cut as posts. I can definitely see why people would want to create the whole page in the editor, rather than create a one-off template and then a one-off page to use that template.

**Comment by webmandesign** [comment]
Just want to point out a related issue I've stumbled on: #52717
